esd_services_api_client/nexus/core/serialization_format.py

```diff
@@ -0,0 +1,83 @@
+"""Serialization format module."""
+from _warnings import warn
+
+import pandas
+from adapta.storage.models.format import (
+    DataFrameParquetSerializationFormat,
+    DictJsonSerializationFormat,
+    SerializationFormat,
+)
+
+
+class SerializationFormatContainer(SerializationFormat):
+    """
+    Serialization format containing multiple formats. The format to use is determined at runtime by the type of the data.
+    """
+
+    def __init__(self, default_serialization_formats: dict = None):
+        self._serialization_formats = (
+            {}
+            if default_serialization_formats is None
+            else default_serialization_formats
+        )
+
+    def __call__(self):
+        return self
+
+    def get_serialization_format(self, data) -> SerializationFormat:
+        """
+        Get the serializer for the data.
+        """
+        return self._serialization_formats[type(data)]()
+
+    def add_serialization_format(
+        self, serialization_format: SerializationFormat
+    ) -> None:
+        """Add a serialization format to the supported formats. Note that only 1 serialization format is allowed per
+        type."""
+
+        serialization_target_type = serialization_format.__orig_bases__[0].__args__[0]
+
+        if serialization_target_type in self._serialization_formats.keys():
+            warn(
```

Comment:
This is tricky since logger is not injected at this pointed (or not guaranteed to be injected). Normally what we'll do in this is raise StartupConfigurationError and fail.

I wonder if we should warn about this at all? Silent replace seems like a logical option, unless you want to go via exception route.

Reply:
I'm ok with removing the warning